### PR TITLE
Reflect Discord's max length message limits

### PR DESCRIPTION
**Pull request by kozaktomas**
Discord is limiting the length of the message (title, description). In this PR, I propose truncating the strings like AM does for pushover or opsgenie.

Doc: https://discord.com/developers/docs/resources/channel#embed-object-embed-limits

Closes: #3310

### CODE REVIEW

**Review by grobinson-grafana — APPROVED** [review]
Nice work!

File: notify/discord/discord.go [hunk 1]
```diff
@@ -90,14 +97,20 @@ func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
...
 	}
 
-	title := tmpl(n.conf.Title)
+	title, truncated := notify.TruncateInRunes(tmpl(n.conf.Title), maxTitleLenRunes)
 	if err != nil {
 		return false, err
 	}
-	description := tmpl(n.conf.Message)
+	if truncated {
+		level.Warn(n.logger).Log("msg", "Truncated title", "incident", key, "max_runes", maxTitleLenRunes)
```
Comment (gotjosh) [inline notify/discord/discord.go:105]:
```suggestion
		level.Warn(n.logger).Log("msg", "Truncated title", "key", key, "max_runes", maxTitleLenRunes)
```

I believe we call this `key` in other notifiers e.g. https://github.com/prometheus/alertmanager/blob/main/notify/slack/slack.go#L116C55-L116C58

File: notify/discord/discord.go [hunk 2]
```diff
@@ -90,14 +97,20 @@ func (n *Notifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
...
+	}
+	description, truncated := notify.TruncateInRunes(tmpl(n.conf.Message), maxDescriptionLenRunes)
 	if err != nil {
 		return false, err
 	}
+	if truncated {
+		level.Warn(n.logger).Log("msg", "Truncated message", "incident", key, "max_runes", maxDescriptionLenRunes)
```
Comment (gotjosh) [inline notify/discord/discord.go:112]:
```suggestion
		level.Warn(n.logger).Log("msg", "Truncated message", "key", key, "max_runes", maxDescriptionLenRunes)
```

ditto

**Review by gotjosh — COMMENTED** [review]
LGTM, but please take a look at my comments.

File: notify/discord/discord.go (hunk 1 above)
Comment (kozaktomas) [reply notify/discord/discord.go]: Fixed in beecbcb

File: notify/discord/discord.go (hunk 2 above)
Comment (kozaktomas) [reply notify/discord/discord.go]: Fixed in beecbcb

**Review by gotjosh — APPROVED** [review]
LGTM

**Pull request merged**